chapters/classes.tex

```diff
@@ -483,81 +538,131 @@ \subsection{Component Variability Prefixes}\label{component-variability-prefixes
+parameter Integer p = length("Hello")
+  annotation(Evaluate = false);                    // Non-evaluable parameter
+parameter Boolean b = false;                       // Evaluable parameter
+
+/* Fulfillment of acyclic binding rule might cause evaluation of n;
+ * to break the cycle, a tool might evaluate either b, n, or both:
+ */
+parameter Real x = if b and n < 3 then 1 - x else 0;
+
+/* Fulfillment of acyclic binding rule cannot cause evaluation of p;
+ * to break the cycle, evaluation of b is the only option:
+ */
+parameter Real y = if b and p < 3 then 1 - y else 0;
 \end{lstlisting}
 \end{example}
 
-The \emph{expression} must be a \lstinline!Boolean! scalar expression, and must be a parameter expression.
-
 \begin{nonnormative}
-A parameter expression is required since it shall be evaluated at compile time.
+For a parameter in a valid model, presence of \lstinline!Evaluate! (\cref{modelica:Evaluate}) makes it possible to tell immediately whether it is an evaluable or non-evaluable parameter, at least as long as the warning described in \cref{modelica:Evaluate} isn't triggered.
+To see this, note that \lstinline!Evaluate = false! makes it a non-evaluable parameter by definition, and that \lstinline!Evaluate = true! would trigger the warning if the parameter is non-evaluable.
 \end{nonnormative}
 
-A redeclaration of a component shall not include a condition attribute;
-and the condition attribute is kept from the original declaration (see
-\cref{interface-compatibility-or-subtyping}).
+\begin{nonnormative}
+With every non-evaluable parameter, there is at least one reason why it isn't an evaluable parameter.
+This information is useful to maintain in tools, as it allows generation of informative error messages when a violation of evaluable expression variability is detected.
+For example:
+\begin{lstlisting}[language=modelica]
+  parameter Integer n =
+    if b then 1 else 2;    // Non-evaluable parameter due to variability of b.
+  parameter Boolean b(fixed = false);
+                           // Non-evaluable parameter due to fixed = false.
+  Real[n] x;               // Variability error: n must be evaluable.
+initial equation
+  b = n > 3;
+\end{lstlisting}
+Here, a good error message for the variability error can include the information that the reason for \lstinline!n! being a non-evaluable parameter is that it has a dependency on the non-evaluable parameter \lstinline!b!.
 
-If the \lstinline!Boolean! expression is false the component (including its modifier) is removed from the flattened DAE, and connections to/from the component are removed.
-A component declared with a condition-attribute can only be modified and/or used in connections.
+(Would evaluable and non-evaluable parameters have had different variability prefixes in their component declarations, it would have been possible to detect and report errors directly at the evaluable component declarations instead.)
```

Comment:
To me this seems like a consideration for a different design that shouldn't be present in the specification.

Reply:
Yeah, I don't remember how it ended up here; maybe it was meant more as a clarification to reviewers than to end users.  I'll remove it.

Reply:
Removed.  Resolving.